internal/command/synthetics/synthetics.go

```diff
@@ -0,0 +1,44 @@
+package synthetics
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/metrics/synthetics"
+)
+
+func New() *cobra.Command {
+	const (
+		short = "Synthetic monitoring"
+		long  = `Synthetic monitoring management.`
+	)
+	cmd := command.New("synthetics", short, long, nil)
+	cmd.AddCommand(
+		newAgent(),
+	)
+	return cmd
+}
+
+func newAgent() *cobra.Command {
+	const (
+		short = "Runs the Synthetics agent"
+		long  = "Runs the Synthetics agent in the foreground."
+	)
+	cmd := command.New("agent", short, long, runAgent,
+		command.RequireSession,
+	)
+	cmd.Args = cobra.NoArgs
+	return cmd
+}
+
+func runAgent(ctx context.Context) (err error) {
+	err = synthetics.RunAgent(ctx)
+	if err != nil {
+		return err
+	}
+	for {
```

Comment:
It's doing an active wait to run the command in the foreground as `RunAgent` is returning immediately (it's executed as a background task when is run as part of the agent)
 
I'm just starting with Go, if you have an advice for a better practice to run thing in the fg, they are more than welcome :)

Reply:
Sorry, I'm just saying that you could have

```go
<-ctx.Done()
return ctx.Err()
```

instead of 

```go
for {
  <-ctx.Done()
  return ctx.Err()
}
```

The loop is only ever going to run a single time.

Reply:
> Sorry, I'm just saying that you could have
> 
> ```go
> <-ctx.Done()
> return ctx.Err()
> ```
> 
> instead of
> 
> ```go
> for {
>   <-ctx.Done()
>   return ctx.Err()
> }
> ```
> 
> The loop is only ever going to run a single time.

I must be blind 🤦 😅  Thanks!